devito/dimension.py

```diff
@@ -31,10 +30,38 @@ def __str__(self):
     @property
     def symbolic_size(self):
         """The symbolic size of this dimension."""
-        try:
-            return Number(self.size)
-        except TypeError:
-            return self.rtargs[0].as_symbol
+        return self.rtargs[0].as_symbol
```

Comment:
Suggestion for the future: perhaps we wanna return a ``namedtuple`` when it comes to the ``Dimension``'s ``rtargs``